spec/controllers/v2/reports_controller_spec.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe V2::ReportsController do
+  before do
+    stub_rbac_permissions(
+      Rbac::INVENTORY_HOSTS_READ,
+      Rbac::REPORT_READ
+    )
+  end
+
+  let(:current_user) { FactoryBot.create(:v2_user) }
+  let(:rbac_allowed?) { true }
+
+  before do
+    request.headers['X-RH-IDENTITY'] = current_user.account.identity_header.raw
+    allow(StrongerParameters::InvalidValue).to receive(:new) { |value, _| value.to_sym }
+    allow(controller).to receive(:rbac_allowed?).and_return(rbac_allowed?)
+  end
+
+  context '/reports' do
+    describe 'GET index' do
+      let(:attributes) do
+        {
+          title: :title,
+          os_major_version: :os_major_version,
+          ref_id: :ref_id,
+          description: :description,
+          profile_title: :profile_title,
+          business_objective: :business_objective,
+          all_systems_exposed: :all_systems_exposed,
+          compliance_threshold: :compliance_threshold,
+          assigned_system_count: -> { 1 },
+          compliant_system_count: -> { 1 },
+          result_system_count: -> { 1 },
+          unsupported_system_count: -> { 0 }
```

Comment:
Ideally you could use the same `attributes` for `index` and `show` if you populate the factory properly.